listenbrainz/webserver/views/playlist_api.py

```diff
@@ -152,6 +153,113 @@ def serialize_jspf(playlist: Playlist):
     return {"playlist": pl}
 
 
+def serialize_xspf(playlist: Playlist):
+    """
+        Given a playlist, return a properly formatted dict that can be passed to jsonify.
+    """
+
+    playlist_root = ET.Element("playlist")
+    playlist_root.attrib = {
+        "version": "1",
+        "xmlns": "http://xspf.org/ns/0"
+    }
+
+    creator = ET.SubElement(playlist_root, "creator")
+    creator.text = playlist.creator
+
+    title = ET.SubElement(playlist_root, "title")
+    title.text = playlist.name
+
+    identifier = ET.SubElement(playlist_root, "identifier")
+    identifier.text = PLAYLIST_URI_PREFIX + str(playlist.mbid)
+
+    date = ET.SubElement(playlist_root, "date")
+    date.text = playlist.created.astimezone(datetime.timezone.utc).isoformat()
+
+    if playlist.description:
+        annotation = ET.SubElement(playlist_root, "annotation")
+        annotation.text = playlist.description
+
+    # Extension
+    playlist_extension = ET.SubElement(playlist_root, "extension", application=PLAYLIST_EXTENSION_URI)
+
+    playlist_extension_public = ET.SubElement(playlist_extension, "public")
+    playlist_extension_public.text = playlist.public
```

Comment:
Public is a bool and if that is passed to the XML serializer, it crashes with:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/xml/etree/ElementTree.py", line 1028, in _escape_cdata
    if "&" in text:
       ^^^^^^^^^^^
TypeError: argument of type 'bool' is not iterable

The bool will need to be cast to a string for this to work.